mypy/types.py

```diff
@@ -1162,23 +1162,45 @@ class TypeType(Type):
     # a generic class instance, a union, Any, a type variable...
     item = None  # type: Type
 
+    # this class should not be created directly
+    # use make method which may return either TypeType or UnionType
+    # this is to ensure Type[Union[A, B]] is always represented as Union[Type[A], Type[B]]
     def __init__(self, item: Type, *, line: int = -1, column: int = -1) -> None:
```

Comment:
Yes, right - and I can easily do right now. 

What I meant is that in the future someone might incorrectly call `TypeType(item)`, where `item` is a `Type` that sometimes is `UnionType`, and sometimes another subtype of `Type`. Of course, the correct approach is to do `if isinstance(item, UnionType) ... else ...`, but it's totally not obvious for a contributor unfamiliar with this issue (and even to me in a few months).

That would introduce a hard-to-detect bug: `x` may be a `UnionType` only rarely, so the tests won't catch it. Static typing also won't catch it either because I can't tell `__init__` that `item` can be of any subtype of `Type` except `UnionType`.

So what I was trying to do is enforce correctness by doing `isinstance` check inside `TypeType`; but of course, it can't be done inside `__init__`, so I had to disable `__init__` (to prevent accidental calls to it) and add `__new__`.

I can easily change it though if you think it's a bit too weird.